app/controllers/mainController.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-restricted-globals */
+/* eslint-disable consistent-return */
+// Utilisation de PSQL
+
+const dataMapper = require('../dataMappers/filmDataMapper');
+const userDataMapper = require('../dataMappers/userDataMapper');
+const formatUser = require('../utils/formatUser');
+
+const mainController = {
+
+  renderHomePage(_, res) {
+    res.render('home');
+  },
+
+  redirectHomePage(_, res) {
+    res.redirect('/app');
+  },
+
+  async renderFilmsListPage(_, res) {
+    try {
+      const films = await dataMapper.getdAllFilms();
+      res.render('films', { films });
+    } catch (err) {
+      res.status(500).send('Server Error');
+    }
+  },
+
+  async renderFilmsByReleaseDatePage(req, res) {
+    try {
+      const param = req.params.filmDate;
+
+      if (isNaN(param) === true || param === undefined) {
+        return res.status(404).render('404', { error: `${param} is not allowed.` });
+      }
+      const filmDate = Number(req.params.filmDate);
+      const foundFilms = await dataMapper.getFilmsByDate(filmDate);
+
+      res.render('filmsByDates', { foundFilms });
```

Comment:
```suggestion
      res.render('filmsByDates', { films: foundFilms });
```

Je me demande si tu ne pouvais pas te servir de la même View que la liste des films de base (sans filtrage par année) pour éviter d'avoir 2 pages